vfs/read_write.go

```diff
@@ -260,7 +261,21 @@ func (fh *RWFileHandle) _readAt(b []byte, off int64, release bool) (n int, err e
 		// Do the writing with fh.mu unlocked
 		fh.mu.Unlock()
 	}
-	n, err = fh.item.ReadAt(b, off)
+	for {
+		fh.item.ProtectCache()
+		n, err = fh.item.ReadAt(b, off)
+		fh.item.UnprotectCache()
+		if err == nil {
+			break
+		}
+		log.Trace(fh.logPrefix(), "err=%v err.Error()=%v", &err, err.Error())
+		if !fserrors.IsErrNoSpace(err) && err.Error() != "no space left on device" {
```

Comment:
Likewise this logic should be in Item I think, then you could get rid of `item.GetCache`

Reply:
Agreed!